src/index.js

```diff
@@ -73,6 +73,13 @@ const encodeOptions = {
   }
 }
 
+export const encodeOptions = {
+  ..._encodeOptions,
+  typeEncoders: {
+    ..._encodeOptions.typeEncoders
+  }
+}
+
```

Comment:
Is it worth doing Object.freeze here and nested?

Reply:
Not as important I think; I haven't kept up with perf on Object.freeze, do you know if there's still a penalty for using a frozen object?

Reply:
from what I can find, a perf hit (which had to do with accessing properties) was fixed for v8 in 2014 https://stackoverflow.com/a/29331374. IIRC Object.preventExtensions (which uses a similar code path to Object.freeze in v8) was previously used inside of the CID class before it was removed for the call being too expensive.

In this case there would only be a constant amount of Object.freeze calls but I'm fine not adding them. It would be nice if there were an easy way to make an object completely static. Could also export functions which return copies of the options. What do you think?

Reply:
No, this is fine, there's such a thing as being too strict in JavaScript—part of the beauty of the language comes from being able to hack around pseudo-strictness in novel ways. My concern with the export is having some piece of code mess with it and causing unintended consequences somewhere entirely unrelated because the codec is so low level in the stack and gets used in surprising ways. So as long as we firewall the codec itself from too much hackery then I think this is fine.